src/System.Linq.Expressions/src/System/Linq/Expressions/CatchBlock.cs

```diff
@@ -158,7 +157,20 @@ public static CatchBlock MakeCatchBlock(Type type, ParameterExpression variable,
         {
             ContractUtils.RequiresNotNull(type, nameof(type));
             ContractUtils.Requires(variable == null || TypeUtils.AreEquivalent(variable.Type, type), nameof(variable));
-            if (variable != null && variable.IsByRef)
+            if (variable == null)
+            {
+                TypeUtils.ValidateType(type, nameof(type));
```

Comment:
It seems that this PR and the two other related ones for `Convert` and `Goto` always combine `ValidateType` and the `IsRef` and `IsPointer` checks. The same checks seem to be missing in other places where `ValidateType` is used (e.g. `TypeAs`), which makes me wonder whether we could move the checks into `ValidateType` in the first place. Thoughts?